### DISCUSSION TITLE
Kitty keyboard protocol doesn't work with foot >=v1.17.0

### DISCUSSION THREAD

**Discussion by postsolar** [helix-editor/helix#10374] [Troubleshooting]
### Summary

See https://codeberg.org/dnkl/foot/issues/1642

From what I understand it's a crossterm's responsibility?

### Reproduction Steps

 

I tried this:

0. Have a `keys.normal.s-backspace` mapping
1. `hx`
2. `s-backspace`
3. Doesn't work

I expected this to happen:

Work

Instead, this happened:

No work

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

Linux

### Terminal Emulator

foot version: 1.17.0 +pgo +ime +graphemes -assertions

### Installation Method

flake

### Helix Version

24.03-125-gf031932f

**Comment by the-mikedavis** [comment]
I believe this is a foot bug. Shift+backspace gives a backspace key event with a shift modifier for me on Kitty but not foot, although I was testing against foot 1.14.0. To dig deeper you can use crossterm's event-read example (clone crossterm and `cargo run --example event-read` in the crossterm repo directory). You can modify the flags in that example to match what we select: https://github.com/helix-editor/helix/blob/124576059544aae3ed8c16247ce01c7e1b95d06e/helix-tui/src/backend/crossterm.rs#L161-L164

**Reply by postsolar** [reply]
Yes what I report is consistent with crossterm (given the same flags as Helix) and yes it's definitely foot-specific, but as expressed by @dnkl in https://codeberg.org/dnkl/foot/issues/1642#issuecomment-1660461 it's intended behavior rather than a bug.

It's weird it doesn't work for you on foot 1.14.0 — I use the commit directly before the commit that introduced this problem (so d3b348a5b183b0e6295dc985b1d5dcd939cb5c6e), which would correspond to 1.16.2, and it works correctly for me.

I tried simply adding more flags to Helix, and it did solve the issue at hand, but, not unexpectedly, raised some other critical problem (I don't remember what it was exactly).

**Reply by postsolar** [reply]
Seems to be the same issue with alacritty 0.13.2. Works fine with the wider set of flags, but not with the set of flags used by Helix.
